### PR TITLE
Serve CSV from riak

### PR DESCRIPTION
**Pull request by fkrause98**
## Context:
- The idea is to provide an example of use case for Riak that is not a simple key-value. The code here would be used to set up an example like explained in [here](https://hackmd.io/8BFqFT7IRq-8ZHnaHhtApg?both). The goal is to show case how to use Riak as a way to distribute several files (or one big file) in memory.
### Added RiaxCSV module for tutorial:
  - This change is to provide a function to store a CSV as index -> row pairs distributed among running nodes.
### Update VNode for tutorial:
  - I've added a new handle_command clause to store the file without logging.
### CSV Visualization
- I've added a Liveview and JSON views for the rows of the csv file, where it is shown both the csv's row info and the node that is storing it.

## How to test it:
If you really want to test everything, you can follow this steps:

1.  Download the CSV file from [here](https://www.kaggle.com/datasets/gauravduttakiit/bitcoin-tweets-16m-tweets-with-sentiment-tagged?resource=download%29).
 This is just a CSV of tweets about bitcoin.
2. Run mix deps.get
3. Set up the 4 nodes on different terminals with `make node1_limited`, `make node2_limited`, `make node3_limited` and `make phoenix_limited`. 
4. Join the nodes with `:riak_core.join(node@host)`.
5. Run `Riax.CSV.distribute_csv(path)`, with the argument being a string with the path to the CSV file.
    **Be ware this will probably consume a lot of memory.**
6. Now you should be able to visit: http://localhost:4003/api/num or http://localhost:4003/csv/num, num > 0.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,3 +9,12 @@ node3:
 
 phoenix:
 	MIX_ENV=dev iex --name phoenix@127.0.0.1 -S mix phx.server
+
+node1_limited:
+	MIX_ENV=dev1 iex --erl "+MMsco true +MMscs 2500" --name dev1@127.0.0.1 -S mix run
```
Comment (facundoolano) [inline Makefile:14]: extract these args to a variable, or extract the entire iex command to a separate target o variable, and set the parts that change as variables

File: lib/riax_csv_example.ex [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  by storing it as row_number -> row values mapping.
+  """
+  def distribute_csv(path) do
+    :rpc.multicall(Riax.CSV, :store_csv, [path])
+  end
+
+  @doc """
+  Stores a CSV in the running node's assigned ring partitions.
+  Each of the CSV's rows are stored using the row's index number as key.
+  Eg. If I have "100" as key and the current ring status
```
Comment (facundoolano) [inline lib/riax_csv_example.ex:19]: doesn't sound like the best choice for indexing, if you add an element in the middle all the indexes change?

Comment (facundoolano) [reply lib/riax_csv_example.ex:19]: maybe it's not relevant for the example

Comment (fkrause98) [reply lib/riax_csv_example.ex:19]: What would you suggest instead of indexing? 🤔

Comment (fkrause98) [reply lib/riax_csv_example.ex:19]: For the example is not really relevant, but I could address it.

File: Makefile (hunk 1 above)
Comment (fkrause98) [reply Makefile:14]: Do you mean something like this?
```makefile
MEMORY_LIMIT := "+MMsco true +MMscs 2500"

node1_limited:
	MIX_ENV=dev1 iex --erl $(MEMORY_LIMIT) --name dev1@127.0.0.1 -S mix run
node2_limited:
	MIX_ENV=dev2 iex --erl $(MEMORY_LIMIT) --name dev2@127.0.0.1 -S mix run
node3_limited:
	MIX_ENV=dev3 iex --erl $(MEMORY_LIMIT) --name dev3@127.0.0.1 -S mix run
phoenix_limited:
	MIX_ENV=dev iex --erl $(MEMORY_LIMIT) --name phoenix@127.0.0.1 -S mix phx.server
```

Comment (facundoolano) [reply Makefile:14]: yeah but more like VM_ARGS or ERL_ARGS

File: lib/riax_csv_example.ex (hunk 2 above)
Comment (facundoolano) [reply lib/riax_csv_example.ex:19]: I would use a field in the csv instead of the row number, but don't worry about it for now

**Review by facundoolano — APPROVED** [review]

**Pull request merged**